lib/ansible/modules/cloud/azure/azure_rm_batchaccount.py

```diff
@@ -0,0 +1,307 @@
+        description:
+        - The name of the resource group in which to create the Batch Account.
+        required: true
+    name:
+        description:
+        - The name of the Batch Account.
+        required: true
+    location:
+        description:
+        - Specifies the supported Azure location where the resource exists.
+        required: true
+    auto_storage_account:
+        description:
+        - The ID of the Batch Account auto storage account.
+    key_vault_reference:
+        description:
+        - A reference to the Azure key vault associated with the Batch account.
+        suboptions:
+            id:
+                description:
+                - The resource ID of the Azure key vault associated with the Batch
+                    account.
+                required: true
+            url:
+                description:
+                - The URL of the Azure key vault associated with the Batch account.
+                required: true
+    pool_allocation_mode:
+        description:
+        - The pool acclocation mode of the Batch Account.
+        default: batch_service
+        choices:
+        - batch_service
+        - user_subscription
+    state:
+        description:
+        - Assert the state of the Batch Account.
+        - Use 'present' to create or update a Batch Account and 'absent' to delete
+            it.
+        default: present
+        choices:
+        - present
+        - absent
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Junyi Yi (@JunyiYi)"
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) Batch Account
+    azure_rm_batchaccount:
+      resource_group: MyResGroup
+      name: "test_object"
+      location: West US
+      auto_storage_account: MyStorageAccountId
+      state: present
+'''
+
+RETURN = '''
+id:
+    description:
+    - The identifier of the Batch Account resource.
+    returned: always
+    type: str
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _snake_to_camel
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.polling import LROPoller
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.batch import BatchManagementClient
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMBatchAccount(AzureRMModuleBase):
+    """Configuration class for an Azure RM Batch Account resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                required=True,
+                type='str'
+            ),
+            name=dict(
+                required=True,
+                type='str'
+            ),
+            location=dict(
+                required=True,
```

Comment:
is location required at updating?

Reply:
No. It is only required during creation.